akka-docs/rst/java/code/docs/cluster/FactorialBackend.java

```diff
@@ -1,38 +1,33 @@
 package docs.cluster;
 
 import java.math.BigInteger;
-import java.util.concurrent.Callable;
 import scala.concurrent.Future;
-import akka.actor.UntypedActor;
+import akka.actor.AbstractActor;
 import akka.dispatch.Mapper;
 import static akka.dispatch.Futures.future;
 import static akka.pattern.Patterns.pipe;
 
 //#backend
-public class FactorialBackend extends UntypedActor {
+public class FactorialBackend extends AbstractActor {
 
   @Override
-  public void onReceive(Object message) {
-    if (message instanceof Integer) {
-      final Integer n = (Integer) message;
-      Future<BigInteger> f = future(new Callable<BigInteger>() {
-        public BigInteger call() {
-          return factorial(n);
-        }
-      }, getContext().dispatcher());
-
-      Future<FactorialResult> result = f.map(
+  public Receive createReceive() {
+    return receiveBuilder()
+      .match(Integer.class, n -> {
+        Future<BigInteger> f = future(() -> factorial(n),
+          getContext().dispatcher());
+
+        Future<FactorialResult> result = f.map(
           new Mapper<BigInteger, FactorialResult>() {
```

Comment:
And skip the old `mapper` and use the built in transform method in completion stage instead (can't remember which one of the 500 methods it is from the top of my head though)?